utilities/face_utility.py

```diff
@@ -1,46 +1,72 @@
 import dlib
 import numpy
+import os
 
+from glob import glob
+from skimage import io
+
+from utilities.path_utility import build_path, name_from_path
+
+FACES_DIRECTORY = 'faces';
+FACES_FILENAME = 'faces.npy'
+
+ANALYZED_FACES_PATH = build_path(FACES_DIRECTORY, FACES_FILENAME)
 
 face_detector = dlib.get_frontal_face_detector()
 face_recognition = dlib.face_recognition_model_v1('./models/dlib_face_recognition_resnet_model_v1.dat')
 shape_predictor = dlib.shape_predictor('./models/shape_predictor_68_face_landmarks.dat')
 
-
 def face_to_vector(image, face):
-    return (
-        numpy
-            .array(face_recognition.compute_face_descriptor(image, face))
-            .astype(float)
-    )
+    face_descriptor = face_recognition.compute_face_descriptor(image, face)
+    face_vector = numpy.array(face_descriptor).astype(float)
+
+    return face_vector
+
+def face_from_image(image, face):
+    size = face.height() * face.width()
+    shape = shape_predictor(image, face)
+
+    return size, shape
 
 def faces_from_image(image):
-    upsampling_factor = 0
-    detected_faces = face_detector(image, upsampling_factor)
+    detected_faces = face_detector(image, 0)
+    faces_in_image = [face_from_image(image, face) for face in detected_faces]
+    sorted_faces = [face for _, face in sorted(faces_in_image, reverse=True)]
 
-    faces = [
-        (face.height() * face.width(), shape_predictor(image, face))
-        for face in detected_faces
-    ]
+    return sorted_faces
 
-    return [face for _, face in sorted(faces, reverse=True)]
+def analyze_faces():
```

Comment:
Either document what this function does or rename it to indicate what it actually does, save the results to the disk